test/testprocess.c

```diff
@@ -0,0 +1,175 @@
+    SDL_Process *process = NULL;
+    Sint64 pid;
+    SDL_IOStream *process_stdin = NULL;
+    SDL_IOStream *process_stdout = NULL;
+    const char *text_in = "Tests whether we can write to stdin and read from stdout\r\n{'succes': true, 'message': 'Success!'}\r\nYippie ka yee\r\nEOF";
+    size_t amount_written;
+    size_t amount_to_write;
+    char buffer[128];
+    size_t total_read;
+    SDL_bool wait_result;
+    int exit_code;
+
+    props = SDL_CreateProperties();
+    SDL_SetPointerProperty(props, SDL_PROP_PROCESS_CREATE_ARGS_POINTER, (void *)process_args);
+    SDL_SetPointerProperty(props, SDL_PROP_PROCESS_CREATE_ENVIRONMENT_POINTER, process_env);
+    SDL_SetBooleanProperty(props, SDL_PROP_PROCESS_CREATE_STDIN_BOOLEAN, SDL_TRUE);
+    SDL_SetBooleanProperty(props, SDL_PROP_PROCESS_CREATE_STDOUT_BOOLEAN, SDL_TRUE);
+    process = SDL_CreateProcessWithProperties(props);
+    SDL_DestroyProperties(props);
+    SDLTest_AssertCheck(process != NULL, "SDL_CreateProcessWithProperties()");
+    if (!process) {
+        goto failed;
+    }
+
+    props = SDL_GetProcessProperties(process);
+    SDLTest_AssertCheck(props != 0, "SDL_GetProcessProperties()");
+
+    pid = SDL_GetNumberProperty(props, SDL_PROP_PROCESS_PID_NUMBER, 0);
+    SDLTest_AssertCheck(pid != 0, "Checking process ID, expected non-zero, got %" SDL_PRIs64, pid);
+
+    process_stdin = (SDL_IOStream *)SDL_GetPointerProperty(props, SDL_PROP_PROCESS_STDIN_POINTER, NULL);
+    SDLTest_AssertCheck(process_stdin != NULL, "SDL_GetPointerProperty(SDL_PROP_PROCESS_STDIN_POINTER) returns a valid IO stream");
+    process_stdout = (SDL_IOStream *)SDL_GetPointerProperty(props, SDL_PROP_PROCESS_STDOUT_POINTER, NULL);
+    SDLTest_AssertCheck(process_stdout != NULL, "SDL_GetPointerProperty(SDL_PROP_PROCESS_STDOUT_POINTER) returns a valid IO stream");
+    if (!process_stdin || !process_stdout) {
+        goto failed;
+    }
+    SDLTest_AssertPass("About to write to process");
+    amount_to_write = SDL_strlen(text_in);
+    amount_written = SDL_WriteIO(process_stdin, text_in, amount_to_write);
+    SDLTest_AssertCheck(amount_written == amount_to_write, "SDL_WriteIO(subprocess.stdin) wrote %" SDL_PRIu64 " bytes, expected %" SDL_PRIu64, (Uint64)amount_written, (Uint64)amount_to_write);
+    if (amount_to_write != amount_written) {
+        goto failed;
+    }
+
+    total_read = 0;
+    for (;;) {
+        size_t amount_read;
+        if (total_read >= sizeof(buffer) - 1) {
+            SDLTest_AssertCheck(0, "Buffer is too small for input data.");
+            goto failed;
+        }
+
+        SDLTest_AssertPass("About to read from process");
+        amount_read = SDL_ReadIO(process_stdout, buffer + total_read, sizeof(buffer) - total_read - 1);
+        total_read += amount_read;
+        buffer[total_read] = '\0';
+        if (total_read >= sizeof(buffer) - 1 || SDL_strstr(buffer, "EOF")) {
+            break;
+        }
+        SDL_CPUPauseInstruction();
+    }
```

Comment:
subprocess i/o is currently not blocking. Is this something we want to add/change?

Reply:
I don't think so, but we should document it.